app/state_machines/configurable_state_machine/machine.rb

```diff
@@ -132,10 +181,10 @@ def trigger_event(event:, params:)
       event_config = state_config[event]
       if can_trigger_event?(event_name: event, metadata: params)
         ActiveRecord::Base.transaction do
-          to_state = find_destination_state(event_config: event_config, user: params[:acting_user])
-          to_workflow = find_destination_workflow(event_config: event_config)
+          to_state = find_destination_state(event_config: event_config, user: user)
+          to_workflow = find_destination_workflow(event_config: event_config, user: user)
           CaseTransition.unset_most_recent(@kase)
-          write_transition(event: event, to_state: to_state, to_workflow: to_workflow, params: params)
+          write_transition(event: event, to_state: to_state, to_workflow: to_workflow, params: params.clone)
```

Comment:
Did you mean to re-add the `.clone`? It looks like `write_transition` is cloning the `params` that come in.

Reply:
I'm passing the ```params.clone``` to the named parameter ```params```.

Reply:
I think it was the other way around ... it used to do `params.clone` and I pointed out that the cloning should be part of `write_transition` since it's the one dicking around.

https://github.com/ministryofjustice/correspondence_tool_staff/blob/cea8b21705877dac4c3f201d585be00c07c11d2f/app/state_machines/configurable_state_machine/machine.rb#L322

Our previous discussion:

https://github.com/ministryofjustice/correspondence_tool_staff/pull/105#discussion-diff-173421354L121